apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-steps/domain-step/index.tsx

```diff
@@ -66,7 +82,7 @@ const DomainStep: React.FunctionComponent< LaunchStepProps > = ( { onPrevStep, o
 					onExistingSubdomainSelect={ onExistingSubdomainSelect }
 					analyticsUiAlgo="editor_domain_modal"
 					segregateFreeAndPaid
-					locale={ document.documentElement.lang }
+					locale={ locale }
```

Comment:
@p-jackson Is this change alright? Asking since you previously worked on New Onboarding i18n tasks

Reply:
Yes this looks right to me. Getting the locale from the `useLocale()` hook will work as long as the `<LocaleProvider>` has been initialised further up the tree, which I believe it has here:
- Here for `<FocusedLaunchModal>` https://github.com/Automattic/wp-calypso/blob/cf411dead2cb94491a4f2553952a1dfb26783aaa/packages/launch/src/launch/index.tsx#L58
- Here for `attach-launch-sidebar.tsx` https://github.com/Automattic/wp-calypso/blob/cf411dead2cb94491a4f2553952a1dfb26783aaa/apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-launch-sidebar.tsx#L50

It's been initialised with `window.wpcomEditorSiteLaunch.locale` too, which I think was more reliable than `document.documentElement.lang` too, so that's a good change.

Not related to this PR, but something I noticed was we might want to move the `wp_localize_script` call that adds `window.wpcomEditorSiteLaunch.locale` https://github.com/Automattic/wp-calypso/blob/cf411dead2cb94491a4f2553952a1dfb26783aaa/apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/index.php#L66
It's called in `enqueue_launch_button_script_and_style()`, but maybe it should be in `enqueue_launch_flow_script_and_style()` instead, since the button script doesn't utilise the `locale` global.

Reply:
> Not related to this PR, but something I noticed was we might want to move the `wp_localize_script` call

I just had a look at doing this and it's not as straight forward as I thought. The launch button and launch flow scripts both access globals, and I was going to isolate the globals so `wp_localize_script` only added the globals each script used. However both scripts use the `launchFlow` global so don't know which to attach it to 🤷 

It probably doesn't matter that much, both scripts are always enqueued, just wanted to keep the globals tidy.

Reply:
Thank you for reviewing my initial change and exploring further improvements!

> It probably doesn't matter that much, both scripts are always enqueued, just wanted to keep the globals tidy.

I'm not going to include those changes in this PR, but I'm tagging @yansern since he _may_ have some insights (or a better knowledge of the codebase)

Reply:
We can see both launch button and launch script as one, they are just separated for load optimization.
So yup it doesn't matter much, we can keep the `wp_localize_script` on the launch button handle.